### PR TITLE
Fix exceeded expected timeout

### PR DESCRIPTION
**Pull request by wzshiming**
Signed-off-by: Shiming Zhang <wzshiming@foxmail.com>

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: security/pkg/nodeagent/cache/secretcache.go [hunk 1]
```diff
@@ -433,17 +433,18 @@ func (sc *SecretManagerClient) generateKeyCertFromExistingFiles(certChainPath, k
...
 // if it is not able to read file after timeout.
 func (sc *SecretManagerClient) readFileWithTimeout(path string) ([]byte, error) {
-	retryBackoffInMS := int64(firstRetryBackOffInMilliSec)
+	retryBackoffInMS := time.Duration(firstRetryBackOffInMilliSec)
 	for {
 		cert, err := ioutil.ReadFile(path)
 		if err == nil {
 			return cert, nil
 		}
+		if retryBackoffInMS > totalTimeout {
```
Comment (howardjohn) [inline security/pkg/nodeagent/cache/secretcache.go:442]: `totalTimeout` is the entire time. This checks the interval compared to the total time, which is not correct.

For example, if total time is 4 and interval starts at 1, we will only meet this criteria after 7 (1+2+4) seconds have passed?

**Review by howardjohn — CHANGES_REQUESTED** [review]

File: security/pkg/nodeagent/cache/secretcache.go (hunk 1 above)
Comment (wzshiming) [reply security/pkg/nodeagent/cache/secretcache.go]: Thanks, updated

**Comment by wzshiming** [comment]
PING @howardjohn

**Review by howardjohn — CHANGES_REQUESTED** [review]
I don't think this is right. Its a more complicated to understand, doesn't seem to solve any real problems, and its not as accurate (if ReadFile takes a non-trivial amount of time it can be completely wrong).
I wouldn't like to merge this unless this is fixing a real issue. If it is, please update the description with some more info. Thanks!

**Review by myidpt — APPROVED** [review]

**Pull request closed without merging**